### PR TITLE
{gnome-,}resources: merge, fix up

### PR DESCRIPTION
**Pull request by amaxine**
## Description of changes

Follow up to #272993, #270250

cc @Lukas-Heiligenbrunner @EwuUwe @linsui

I've not checked which of the two licence definition was correct (only vs plus) but this fixes the dmidecode/polkit/systemd deps being missing.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/re/resources/package.nix [hunk 1]
```diff
@@ -48,8 +51,16 @@ stdenv.mkDerivation (finalAttrs: {
...
     gtk4
     libadwaita
+    dmidecode
+    polkit
+    systemd
   ];
 
+  postPatch = ''
+    substituteInPlace src/utils/memory.rs \
+      --replace '"dmidecode"' '"${dmidecode}/bin/dmidecode"'
```
Comment (Lukas-Heiligenbrunner) [inline pkgs/by-name/re/resources/package.nix:61]: Wouldn't the same thing also be necessary for `pkexec` if its not in path?
(https://github.com/nokyan/resources/blob/main/src/utils/memory.rs#L185)

Comment (linsui) [reply pkgs/by-name/re/resources/package.nix:61]: Good catch. I thought it should also be patched.

Comment (linsui) [reply pkgs/by-name/re/resources/package.nix:61]: And maybe it doesn't need to be in the buildInputs? What is systemd for?

Comment (linsui) [reply pkgs/by-name/re/resources/package.nix:61]: https://github.com/nokyan/resources/blob/504398b6de2f37fb127217ce4c9850318f42bb5d/src/utils/cpu.rs#L66 This should also be patched.

Comment (amaxine) [reply pkgs/by-name/re/resources/package.nix:61]: I've taken the changes from the other package, but I assume systemd is for something udev related? Upstream references it as a dependency, but I've not looked further.

Agree on the other two changes, will add those to this fixup.

Comment (amaxine) [reply pkgs/by-name/re/resources/package.nix:58]: I've pushed a change adding util-linux for lscpu, and dropping polkit and systemd as:
- makes no sense to pull polkit in if it's not enabled on the system, and the software should fail if it's not present in the specified path (see anything else that calls pkexec in nixpkgs)
- only references to systemd in upstream don't actually make any sense: https://github.com/search?q=repo%3Anokyan%2Fresources%20systemd&type=code

File: pkgs/by-name/re/resources/package.nix [hunk 2]
```diff
@@ -48,8 +50,19 @@ stdenv.mkDerivation (finalAttrs: {
     glib
     gtk4
     libadwaita
+    dmidecode
+    util-linux
   ];
```
Comment (linsui) [inline pkgs/by-name/re/resources/package.nix:55]: They are only runtime deps so we don't need to add them here.

Comment (amaxine) [reply pkgs/by-name/re/resources/package.nix:55]: That's why they're in buildInputs - they are needed at run-time.

**Review by EwuUwe — APPROVED** [review]

File: pkgs/by-name/re/resources/package.nix (hunk 2 above)
Comment (linsui) [reply pkgs/by-name/re/resources/package.nix:55]: Run time deps doesn't need to be in buildInputs, only build time deps need to be here. We make sure the runtime deps are available via the patches.

Comment (Lukas-Heiligenbrunner) [reply pkgs/by-name/re/resources/package.nix:55]: @amaxine I guess you forgot about this one. Could you remove the two lines please and then I think this one can be merged. :)

Comment (amaxine) [reply pkgs/by-name/re/resources/package.nix:55]: Nope, just been on a very away from any computers holiday :-) I'll finish it up soon-ish.

**Comment by amaxine** [comment]
I'd describe that as out of scope for what this PR is doing, but sure.

**Review by linsui — APPROVED** [review]

**Review by EwuUwe — APPROVED** [review]

**Review by Lukas-Heiligenbrunner — APPROVED** [review]

**Pull request merged**